config/configtls/configtls.go

```diff
@@ -229,18 +229,82 @@ func (c TLSClientSetting) LoadTLSConfig() (*tls.Config, error) {
 	return tlsCfg, nil
 }
 
+// clientCAsReloader is a wrapper object for certificate reloading
+// Its GetClientCAs method will either return the current client CAs or reload from disk
+// if the last reload happened more than ReloadInterval ago
+type clientCAsReloader struct {
+	// Path to the client CAs
+	ClientCAsFile string
+	// ReloadInterval specifies the duration after which the certificate will be reloaded
+	// If not set, it will never be reloaded (optional)
+	ReloadInterval time.Duration
+	nextReload     time.Time
+	certPool       *x509.CertPool
+	lock           sync.RWMutex
+	c              *TLSServerSetting
```

Comment:
We already have a reload mechanism, what does this bring?

Reply:
https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configtls/configtls.go#L105

Reply:
Hi @bogdandrutu ! 

I'll gladly admit that this is a blatant copy of `certReloader`, but I found that the reloading of `client_ca_file` and `cert_file/key_file` are different enough concerns to have their own reloaders.
One first one is only related to server, the other one is related to both client and server, their lifecycles differ, etc...
Please let me know what would be a better approach!

Reply:
@bogdandrutu  meanwhile can you allow the CI to run, that would be a great feedback for us!